File: src/exemplos/05_EstruturasDeDados/06_Arquivo/02-fseek.c
/**      @file: 02-fseek.c
 *     @author: Guilherme N. Ramos (gnramos@unb.br)
 * @disciplina: Algoritmos e Programação de Computadores
 *
 * Exemplo da utilização da função fseek. */

#include <stdio.h>
#include <stdlib.h>

int main() {
    char arquivo[] = "apc.c.bin";
    const char string[] = "Disse o corvo, \"Nunca mais\".\n";
    FILE *fp = fopen(arquivo, "wb+" );

    if(!fp) {
        printf("Não foi possível abrir \"%s\".\n", arquivo);
        return EXIT_FAILURE;
    }

    fwrite(string, sizeof(string), 1, fp);
    fseek(fp, 16, SEEK_SET);
    fwrite("Chega", sizeof(char), 5, fp);
    fclose(fp);

    /* O que você espera que tenha acontecido? Abra o arquivo gerado e confira... */

    return EXIT_SUCCESS;
}

File: src/exemplos/05_EstruturasDeDados/06_Arquivo/02-fseek.py
#  -*- coding: utf-8 -*-
#    @package: 02-fseek.py
#     @author: Guilherme N. Ramos (gnramos@unb.br)
# @disciplina: Algoritmos e Programação de Computadores
#
# Exemplo da utilização da função seek.


if __name__ == '__main__':
    arquivo = 'apc.python.bin'

    string = 'Disse o corvo, \'Nunca mais\'.\n'

    try:
        with open(arquivo, 'wb') as f:
            f.write(string)
            f.seek(16, 0)
            f.write('Chega')

            # O que você espera que tenha acontecido? Abra o arquivo gerado e
            # confira...
    except:
        print('Não foi possível abrir \'%s\'.\n' % arquivo)
        exit(1)
